src/screens/Customers/CustomerMain.tsx

```diff
@@ -1,29 +1,130 @@
 import React, { useEffect, useState } from 'react';
-import { CustomerRecord } from '../../lib/airtable/interface';
-import { getAllCustomers } from '../../lib/airtable/request';
+import { CustomerRecord, SiteRecord } from '../../lib/airtable/interface';
 import { Link, RouteComponentProps } from 'react-router-dom';
+import { store } from '../../lib/redux/store';
+import { createStyles, FormControl, FormHelperText, MenuItem, Select, Theme, withStyles } from '@material-ui/core';
+import UserSearchBar from '../../components/UserSearchBar';
+import CustomerCard from '../../components/CustomerCard';
+import TrieTree from '../../lib/utils/TrieTree';
 
-function CustomerMain(props: RouteComponentProps) {
+const styles = (theme: Theme) =>
+  createStyles({
+    title: {
+      width: '153.9px',
+      height: '33.83px',
+      left: '25.6px',
+      top: '96px',
+      fontFamily: 'Helvetica Neue',
+      fontStyle: 'normal',
+      fontWeight: 500,
+      fontSize: '30px',
+      lineHeight: '115%',
+      color: '#828282',
+      flexGrow: 1,
+    },
+    headerWrapper: {
+      display: 'flex',
+      flexDirection: 'row',
+      width: '100%',
+      paddingTop: '86px',
+      paddingRight: '21px',
+      paddingLeft: '21px',
+    },
+    selectionHeader: {
+      display: 'flex',
+      flexDirection: 'column',
+      flexGrow: 1,
+    },
+    scrollDiv: {
+      maxHeight: '380px',
+      overflow: 'auto',
+      marginLeft: 'auto',
+      marginRight: 'auto',
+      width: '310.71px',
+    },
+  });
+
+interface UserProps {
+  classes: { title: string; headerWrapper: string; selectionHeader: string; scrollDiv: string };
+}
+
+function CustomerMain(props: RouteComponentProps & UserProps) {
   useEffect(() => {
     getCustomers();
   }, []);
 
+  const { classes } = props;
   const [customers, setCustomers] = useState<CustomerRecord[]>([]);
+  const [allCustomersTrie, setAllCustomersTrie] = useState<TrieTree<CustomerRecord>>(new TrieTree('name'));
+  const [fullCustomers, setFullCustomers] = useState<CustomerRecord[]>([]);
```

Comment:
Is this `fullCustomers` used anywhere? We might be able to get away with just `customers` (maybe `filteredCustomers` or some indication that this is the customers being shown is a better name), and the `allCustomersTrie` which will always hold all the customers afaik

Reply:
I forgot that the trie holds all the customers. Thanks!

Reply:
I remember why we needed full customers. There is no way to get the full customers from the trie. With an empty string '' the trie returns nothing. Looked fpr documentation and it seems like you cant get it all back. I looked up how the trie was used in Unloop and we used the same method (keep a full copy of list): 
https://github.com/calblueprint/unloop/blob/10b27cbc70fbc85d2322db00d3fc7088facd2713/app/javascript/components/StaffDashboard/index.js#L45

Reply:
Sounds good, thanks for checking it! :^)